### PR TITLE
Update datasource package_provides_command

### PR DESCRIPTION
**Pull request by bfahr**
### All Pull Requests:

Check all that apply:

* [X] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [X] Is this PR an enhancement?

### Complete Description of Additions/Changes:
* Move datasource to separate module
* Update to fix issues in collecting data
* Add testing for datasource and functions
* Update docs

Signed-off-by: Bob Fahr <20520336+bfahr@users.noreply.github.com>

### CODE REVIEW

**Comment by bfahr** [comment]
This datasource needs a full path to the binary to work properly.  I'm going to add a new `ps` parser to get that information.  So this needs to wait on that parser.

File: insights/specs/datasources/package_provides.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    Arguments:
+        broker: the broker object for the current session
+
+    Returns:
+        DatasourceProvider: Returns the collected information as a file with 1 line per command
+
+    Raises:
+        SkipComponent: Raised if no data is collected
+    """
+    commands = get_filters(Specs.package_provides_command)
```
Comment (xiangce) [inline insights/specs/datasources/package_provides.py:69]: 👍🏻

**Comment by bfahr** [comment]
PR #3121 adds the new `ps` parser will collect the full path to the command which is necessary for this PR.

**Review by xiangce — APPROVED** [review]
Looks good to me

**Comment by patchkez** [comment]
test this please

**Pull request merged**